modules/search-pipeline-common/src/main/java/org/opensearch/search/pipeline/common/helpers/SearchResponseUtil.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.search.pipeline.common.helpers;
+
+import org.opensearch.action.search.SearchResponse;
+import org.opensearch.action.search.SearchResponseSections;
+import org.opensearch.search.SearchHits;
+import org.opensearch.search.profile.SearchProfileShardResults;
+
+/**
+ * Helper methods for manipulating {@link SearchResponse}.
+ */
+public final class SearchResponseUtil {
+    private SearchResponseUtil() {
+
+    }
+
+    /**
+     * Construct a new {@link SearchResponse} based on an existing one, replacing just the {@link SearchHits}.
+     * @param newHits new search hits
+     * @param response the existing search response
+     * @return a new search response where the search hits have been replaced
+     */
+    public static SearchResponse replaceHits(SearchHits newHits, SearchResponse response) {
+        return new SearchResponse(
+            new SearchResponseSections(
```

Comment:
Shouldn't we use `InternalSearchResponse` here?

https://github.com/opensearch-project/OpenSearch/blob/61c5f173b9b6dba7de84423be5a9db283e25291b/server/src/main/java/org/opensearch/action/search/SearchResponse.java#L96C1-L96C1

Reply:
Yes, that would be safer -- my understanding is that `InternalSearchResponse` implements `Writeable`, which is required for the whole `SearchResponse` to be `Writeable`.

While the YamlRestTests succeed with this implementation, I think it would run into problems if called from a transport client. It would probably break cross-cluster search.

I'll make that change in my next commit.